src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/TermAdminController.cs

```diff
@@ -90,7 +89,9 @@ public class TermAdminController : Controller, IUpdateModel {
 
                     foreach (var entry in checkedEntries) {
                         var term = _taxonomyService.GetTerm(entry.Id);
-                        _taxonomyService.DeleteTerm(term);
+                        if (term != null) {
+                            _taxonomyService.DeleteTerm(term);
```

Comment:
When I deleted all the terms on the admin panel via UI got an exception because some of them were null at this point since I deleted the parent first so it was already removed and term was null.